### PR TITLE
Add support for Linux respecting the same API

### PR DESCRIPTION
**Pull request by atamakahere-git**
- Moved OSX implementation from lib.rs to `osx.rs`
- Made a Linux sub module with all utilities within

### CODE REVIEW

**Comment by radiohertz** [comment]
Add Linux CI too pls @tanveerraza789

**Comment by atamakahere-git** [comment]
Done @radiohertz

**Comment by radiohertz** [comment]
i just tested it and it looks broken on osx

File: src/osx.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    impl MediaState {
+        pub fn new() -> Self {
+            Self
+        }
+
+        pub fn mic_status(&self) -> Option<State> {
+            let mut status = -1;
+
+            if unsafe { IsMicrophoneOn(&mut status as _) } != -1 {
```
Comment (radiohertz) [inline src/osx.rs]: it should be 0. why did this change?

File: src/osx.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                return None;
+            }
+
+            Some(if status == -1 { State::Off } else { State::On })
+        }
+
+        pub fn cam_state(&self) -> Option<State> {
+            let mut status = -1;
+
+            if unsafe { IsCameraOn(&mut status as _) } != -1 {
```
Comment (radiohertz) [inline src/osx.rs]: same here

File: src/osx.rs (hunk 1 above)
Comment (atamakahere-git) [reply src/osx.rs]: I did not (at least intentionally), idk how it got changed. weird. Let me fix that

**Review by radiohertz — APPROVED** [review]
LGTM

**Pull request merged**